grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -879,13 +879,14 @@ class InventoryService implements ApplicationContextAware {
     }
 
     List getItemQuantityByBinLocation(Location location, List<InventoryItem> inventoryItems) {
-        List transactionEntries = getTransactionEntriesByInventoryAndInventoryItems(location?.inventory, inventoryItems)
+        List products = inventoryItems.collect { it.product }
+        List transactionEntries = getTransactionEntriesByInventoryAndProduct(location?.inventory, products)
```

Comment:
Now that we're return ALL transaction entries at the product level, we need to filter the resulting bin locations by inventoryItem.
``` 
// pseudocode you might need to use a different mechanism to filter by inventory item 
binLocations = binLocations.findAll { it.inventoryItem in inventoryItems }
```